src/Unity.Mathematics.Editor/PrimitiveVectorDrawer.cs

```diff
@@ -89,8 +89,39 @@ public override void OnGUI(Rect position, SerializedProperty property, GUIConten
             if (attribute is DoNotNormalizeAttribute && string.IsNullOrEmpty(label.tooltip))
                 label.tooltip = Content.doNotNormalizeTooltip;
 
-            EditorGUI.BeginProperty(position, label, property);
-            EditorGUI.MultiPropertyField(position, subLabels, property.FindPropertyRelative(startIter), label);
+            label = EditorGUI.BeginProperty(position, label, property);
+
+            {
+                // This code is basically EditorGUI.MultiPropertyField(Rect, GUIContent[], SerializedProperty, GUIContent), but with the
+                // property visibility assumed to be "All" instead of "OnlyVisible". We really want to have "All" because
+                // it's possible for someone to hide something in the inspector with [HideInInspector] but then manually
+                // draw it themselves later. In this case, if you called EditorGUI.MultiPropertyField() directly, you'd
+                // end up with some fields that point to some unrelated visible property.
+                position = EditorGUI.PrefixLabel(position, label);
+                position.height = 18f;
```

Comment:
Should it be `EditorGUIUtility.singleLineHeight`?
Am I also right to assume none of these needs to display on 2 lines when `EditorGUIUtility.wideMode` is true?

Reply:
1. It's probably good to check if it's just `EditorGUIUtility.singleLineHeight`, or if it's also `+ EditorGUIUtility.standardVerticalSpacing`
2. Wide mode is when the Inspector has passed a certain width. So when the user has a small monitor or otherwise makes the Inspector narrow, things like the `Rect` control will go to 2 lines

Reply:
internal :(

<img width="942" alt="image" src="https://user-images.githubusercontent.com/3121968/136466929-b7af5d06-2620-49af-a09e-3d13534e32b6.png">

Reply:
`kSingleLineHeight` is exposed via [this API](https://docs.unity3d.com/ScriptReference/EditorGUIUtility-singleLineHeight.html). (cf [here](https://github.com/Unity-Technologies/UnityCsReference/blob/61f92bd79ae862c4465d35270f9d1d57befd1761/Editor/Mono/EditorGUIUtility.cs#L564))

Reply:
Fixed, thanks for that one!